diff --git a/include/fuse_common.h b/include/fuse_common.h
index 93f6e48..6e607bd 100644
--- a/include/fuse_common.h
+++ b/include/fuse_common.h
@@ -513 +513,6 @@ struct fuse_loop_config {
-#define FUSE_CAP_OVER_IO_URING (1UL << 31)
+#define FUSE_CAP_OVER_IO_URING (1UL << 31)
+
+/**
+ * Indicates that creation of idmapped mounts is allowed.
+ */
+#define FUSE_CAP_ALLOW_IDMAP (1ULL << 32)
diff --git a/lib/fuse_lowlevel.c b/lib/fuse_lowlevel.c
index ea5994a..c90972b 100644
--- a/lib/fuse_lowlevel.c
+++ b/lib/fuse_lowlevel.c
@@ -2748 +2748,3 @@ static void _do_init(fuse_req_t req, const fuse_ino_t nodeid, const void *op_in,
-			se->conn.capable_ext |= FUSE_CAP_OVER_IO_URING;
+			se->conn.capable_ext |= FUSE_CAP_OVER_IO_URING;
+		if (inargflags & FUSE_ALLOW_IDMAP)
+			se->conn.capable_ext |= FUSE_CAP_ALLOW_IDMAP;
@@ -2905 +2907,3 @@ static void _do_init(fuse_req_t req, const fuse_ino_t nodeid, const void *op_in,
-	}
+	}
+	if (se->conn.want_ext & FUSE_CAP_ALLOW_IDMAP)
+		outargflags |= FUSE_ALLOW_IDMAP;
